mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -1351,8 +1353,14 @@ def Rock_ThreadwiseCopyOp :
   }];
 
   let assemblyFormat = [{
-    attr-dict $source  `->` $dest `:` type($source) `->` type($dest)
+    attr-dict $source (`[` $extraIndicesSource^ `]`)? `->` $dest (`[` $extraIndicesDest^ `]`)? `:` type($source) `->` type($dest)
   }];
+
+  let builders = [
+    // Custom builder to not specify the extra indices
+    OpBuilder<(ins "Value":$source, "Value":$dest, CArg<"bool","false">:$forceUnroll, CArg<"bool","false">:$useIndexDiffs),
+                    [{ build($_builder, $_state, source, {}, dest, {}, forceUnroll, useIndexDiffs); }]>
+  ];
```

Comment:
This reminds me, do we want to, at some point, fix up threadwise_write_all to not take a `length` parameter but to instead take a slice of the input?

Reply:
You mean something like this https://github.com/ROCmSoftwarePlatform/rocMLIR-internal/issues/1260 ?